src/file_parsing.jl

```diff
@@ -1,83 +1,327 @@
 using TOML
+using DocStringExtensions
 
+export ParamDict
+export parse_toml_file,
+    get_parametric_type,
+    iterate_alias,
+    log_component!,
+    get_values,
+    get_parameter_values!,
+    get_parameter_values,
+    check_override_parameter_usage,
+    write_log_file,
+    log_parameter_information,
+    merge_override_default_values,
+    create_parameter_struct
+        
+    
+"""
+    ParamDict{FT}
+
+structure to hold information read-in from TOML file, as well as a parametrization type `FT`
+# Constructors
+
+    ParamDict(data::Dict, dict_type::String, override_dict::Union{Nothing,Dict})
+
+# Fields
+
+$(DocStringExtensions.FIELDS)
+
+"""
+struct ParamDict{FT}
+    "dictionary representing a default/merged parameter TOML file"
+    data::Dict
+    "string to determine how dictionary look-up is performed"
+    dict_type::String
+    "either a nothing, or a dictionary representing an override parameter TOML file"
+    override_dict::Union{Nothing,Dict}
+end
+
+"""
+    parse_toml_file(filepath)
+
+use a TOML parser to read TOML file at `filepath`.
+"""
 function parse_toml_file(filepath)
     return TOML.parsefile(filepath)
 end
 
+"""
+    get_parametric_type(::ParamDict{FT}) where {FT}
 
+obtains the type `FT` from `ParamDict{FT}`.
+"""
+get_parametric_type(::ParamDict{FT}) where {FT} = FT
```

Comment:
```suggestion
float_type(::ParamDict{FT}) where {FT} = FT
```
And fix docs accordingly

Reply:
Oh wait, is this not always the float type?

Reply:
It will be floats. Currently we have not had need for other types (except for arrays of floats) 

(PS I'll address your other comments in due course!)